### PR TITLE
libremidi: init at unstable-2023-05-05

### PR DESCRIPTION
**Pull request by paveloom**
###### Description of changes

https://github.com/jcelerier/libremidi

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 232578` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>libremidi</li>
  </ul>
</details>

**Review by Janik-Haag — APPROVED** [review]
LGTM

File: pkgs/development/libraries/libremidi/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+stdenv.mkDerivation {
+  pname = "libremidi";
+  version = "unstable-2023-05-05";
+
+  src = fetchFromGitHub {
+    owner = "jcelerier";
+    repo = "libremidi";
+    rev = "cd2e52d59c8ecc97d751619072c4f4271fa82455";
+    sha256 = "sha256-CydoCprxqDl5FXjtgT+AckaRTqQAlCDwwrnPDK17A6o=";
```
Comment (natsukium) [inline pkgs/development/libraries/libremidi/default.nix:19]:
```suggestion
    hash = "sha256-CydoCprxqDl5FXjtgT+AckaRTqQAlCDwwrnPDK17A6o=";
```

File: pkgs/development/libraries/libremidi/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ ]
+    ++ lib.optional stdenv.isLinux alsa-lib
```
Comment (natsukium) [inline pkgs/development/libraries/libremidi/default.nix:25]:
```suggestion
  buildInputs = lib.optional stdenv.isLinux alsa-lib
```

File: pkgs/development/libraries/libremidi/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    ++ lib.optionals stdenv.isDarwin [ CoreAudio CoreFoundation CoreMIDI ];
```
Comment (natsukium) [inline pkgs/development/libraries/libremidi/default.nix:26]: One item per line is preferred.

**Review by natsukium — CHANGES_REQUESTED** [review]
Something is wrong with `x86_64-darwin`.
It may be necessary to mark it as broken.

**Comment by paveloom** [comment]
Missing `CoreServices` on Darwin, added them.

**Review by natsukium — APPROVED** [review]
Both `aarch64-darwin` and `x86_64-darwin` seem to be fine.

**Pull request merged**